src/main/java/technology/dice/dicefairlink/AuroraReadonlyEndpoint.java

```diff
@@ -5,236 +5,43 @@
-            }
-          }
-        } catch (Exception ex) {
-          LOGGER.log(
-              Level.SEVERE,
-              String.format(
-                  "Got exception when processing [%s] member of [%s]. Skipping.",
-                  readReplica.getDBInstanceIdentifier(), clusterId),
-              ex);
-        }
-      }
-      return urls;
-    }
-
-    @Override
-    public void run() {
-      try {
-        Optional<DBCluster> dbClusterOptional = this.describeCluster();
-        if (!dbClusterOptional.isPresent()) {
-          LOGGER.log(
-              Level.WARNING,
-              String.format(
-                  "Could not retrieve cluster information for cluster [%s]. Will fallback to [%s] until individual members can be retrieved again",
-                  clusterId, readOnlyEndpoint));
-          return;
-        }
-        List<String> readerUrls =
-            dbClusterOptional.map(cluster -> replicaMembersOf(cluster)).orElse(new ArrayList<>(0));
-        if (replicas.hasSameContent(readerUrls)) {
-          return;
-        }
-        if (readerUrls.isEmpty()) {
-          LOGGER.log(
-              Level.WARNING,
-              "No read replicas found for cluster [{0}]. Will fallback to [{1}] until individual members can be retrieved again",
-              new Object[] {clusterId, readOnlyEndpoint});
-        }
-        replicas = RandomisedCyclicIterator.of(readerUrls);
-        if (LOGGER.isLoggable(Level.FINE)) {
-          LOGGER.log(
-              Level.FINE,
-              String.format(
-                  "Retrieved [%s] read replicas for cluster id [%s] with. List will be refreshed in [%s] seconds",
-                  readerUrls.size(), clusterId, pollerInterval.getSeconds()));
-        }
-      } catch (Exception e) {
-        LOGGER.log(
-            Level.SEVERE,
-            String.format(
-                "Exception while refreshing list of read replicas from cluster [%s]. Skipping",
-                clusterId),
-            e);
-      }
-    }
-
-    public void init() {
-      Optional<DBCluster> dbClusterOptional = this.describeCluster();
-      if (!dbClusterOptional.isPresent()) {
-        throw new RuntimeException(
-            String.format(
-                "Could not find exactly one cluster with cluster id [%s]", this.clusterId));
-      }
-      DBCluster cluster = dbClusterOptional.get();
-      readOnlyEndpoint = cluster.getReaderEndpoint();
-      List<String> readerUrls = replicaMembersOf(cluster);
-      replicas = RandomisedCyclicIterator.of(readerUrls);
-      LOGGER.log(
-          Level.INFO,
-          String.format(
-              "Initialized driver for cluster id [%s] with [%s] read replicas. List will be refreshed every [%s] seconds",
-              clusterId, readerUrls.size(), pollerInterval.getSeconds()));
+    String nextReplica = replicas.next();
+    LOGGER.finer("Obtained replica: " + nextReplica);
+    if (nextReplica != null && nextReplica.equals(lastReplica.get())) {
```

Comment:
Not sure how did it worked before, but looks like here we must take into account case when number of read replicas = 1 and because of that skip the check for the same replica returned more than once sequentially.

Reply:
With only when replica it is supposed to always return the same. It would `next()` twice each time, so I added a guard to only do it if there's more than one